src/include/cache.i

```diff
@@ -68,6 +79,21 @@ __wt_cache_read_gen_new(WT_SESSION_IMPL *session, WT_PAGE *page)
 }
 
 /*
+ * __wt_cache_stuck --
+ *      Indicate if the cache is stuck (i.e., not making progress).
+ */
+static inline bool
+__wt_cache_stuck(WT_SESSION_IMPL *session)
+{
+	WT_CACHE *cache;
+
+	cache = S2C(session)->cache;
+	return (cache->evict_aggressive_score == WT_EVICT_SCORE_MAX &&
```

Comment:
That sounds fine to me: this code has evolved while this branch has been open.

One question for you: the intent of the timing was that eviction would get aggressive after 100ms of not making progress and stuck after 1s.  Leaving the implementation details aside, do you think those times are reasonable?